backends/dpdk/dpdkHelpers.cpp

```diff
@@ -134,27 +134,56 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
                 src1Op = src1Member;
             }
         }
+        if (src2Op->is<IR::Constant>()) {
+            if (auto tb = src2Op->type->to<IR::Type_Bits>()) {
+                if (tb != nullptr && tb->isSigned) {
+                    isSignExtended = true;
+                    auto consOrgBitwidth = src2Op->to<IR::Constant>()->type->width_bits();
```

Comment:
Not checking byte aligned here?We are promoting only for fields which are not byte aligned in original P4, right?

Reply:
Okay. will add a check here for same